eclass/zig-utils.eclass

```diff
@@ -0,0 +1,527 @@
+
+	local zig_supported_versions=(
+		"9999"
+		"0.13.1"
+		"0.13.0"
+	)
+
+	local base_path="${BROOT}/usr/bin"
+
+	local selected_path selected_ver
+	for selected_ver in "${zig_supported_versions[@]}"; do
+		# Check if candidate satisfies ZIG_SLOT condition.
+		if [[ "${selected_ver}" != "${ZIG_SLOT}"* ]]; then
+			continue
+		fi
+
+		# Prefer "dev-lang/zig" over "dev-lang/zig-bin"
+		local candidate_path
+		for candidate_path in "${base_path}/"zig{,-bin}"-${selected_ver}"; do
+			if [[ -x "${candidate_path}" ]]; then
+				selected_path="${candidate_path}"
+				break 2
+			fi
+		done
+	done
+
+	if [[ -z "${selected_path}" ]]; then
+		die "Could not find (suitable) Zig at \"${base_path}\""
+	fi
+
+	declare -g ZIG_EXE="${selected_path}"
+	declare -g ZIG_VER="${selected_ver}"
+	# Sanity check, comment from upstream:
+	# > Check libc++ linkage to make sure Zig was built correctly,
+	# > but only for "env" and "version" to avoid affecting the
+	# > startup time for build-critical commands
+	# > (check takes about ~10 μs)
+	"${ZIG_EXE}" version > /dev/null ||
+		die "Sanity check failed for \"${ZIG_EXE}\""
+}
+
+# @FUNCTION: zig-utils_setup
+# @DESCRIPTION:
+# Checks if running Linux kernel version is supported by Zig.
+# Populates ZIG_TARGET, ZIG_CPU, ZIG_EXE and ZIG_VER global
+# variables with detected values, or, if user set them already,
+# leaves as-is.
+zig-utils_setup() {
+	# Should be first because it sets ZIG_VER which might be used
+	# in the future when setting ZIG_TARGET and ZIG_CPU variables
+	# for incompatible versions.
+	if [[ -z "${ZIG_EXE}" ]]; then
+		zig-utils_find_installation
+	fi
+
+	: "${ZIG_CPU:=$(zig-utils_c_env_to_zig_cpu "${CHOST}" "${CFLAGS}")}"
+	if tc-is-cross-compiler; then
+		: "${ZIG_TARGET:=$(zig-utils_c_env_to_zig_target "${CHOST}" "${CFLAGS}")}"
+	else
+		: "${ZIG_TARGET:=native}"
+	fi
+	declare -g ZIG_CPU ZIG_TARGET
+
+	einfo "ZIG_EXE:    \"${ZIG_EXE}\""
+	einfo "ZIG_VER:     ${ZIG_VER}"
+	einfo "ZIG_TARGET:  ${ZIG_TARGET}"
+	einfo "ZIG_CPU:     ${ZIG_CPU}"
+}
+
+# @FUNCTION: ezig
+# @USAGE: [<args>...]
+# @DESCRIPTION:
+# Runs ZIG_EXE with supplied arguments.  Dies if ZIG_EXE is not set or
+# if command exits with error.  Respects `nonfatal`.
+#
+# Always disables progress tree.  By default enables ANSI escape codes
+# (colours, etc.), user can set NO_COLOR environment variable to
+# disable them.
+ezig() {
+	# Sync description above and comments below with upstream's
+	# "std.io.tty.detectConfig".
+	debug-print-function "${FUNCNAME[0]}" "${@}"
+
+	if [[ -z "${ZIG_EXE}" ]] ; then
+		die "${FUNCNAME[0]}: ZIG_EXE is not set. Was 'zig-utils_setup' called before using ezig?"
+	fi
+
+	# Progress tree is helpful indicator in TTY, but unfortunately
+	# they make Portage logs harder to read in plaintext. We pass
+	# "TERM=dumb" here to have clean logs, and CLICOLOR_FORCE to
+	# preserve colors.
+	# User's NO_COLOR takes precendence over this.
+	TERM=dumb CLICOLOR_FORCE=1 "${ZIG_EXE}" "${@}" ||
```

Comment:
```suggestion
	local -x TERM=dumb
	local -x CLICOLOR_FORCE=1
	"${ZIG_EXE}" "${@}" ||
```

Then you could also make the comments shorter and put them above every line separately.

Reply:
Didn't know this, thanks!

Reply:
1d9c23b3a9cc68c38d47c708f1a5861ab576ca51 (should I choose color or colour spelling? `emerge` has `--color` but `eselect` has `--colour`)

Reply:
We generally go for the reformed spelling, i.e. "color".